src/components/dialog/dialog.scss

```diff
@@ -45,13 +45,18 @@
     }
     .mdc-dialog__container {
         height: 100%;
+        width: var(--dialog-width, auto);
     }
 
     .mdc-dialog__surface {
         width: var(--dialog-width, auto);
         height: var(--dialog-height, auto);
         background-color: var(--dialog-background-color);
         box-shadow: var(--shadow-depth-64);
+
+        max-width: var(--dialog-max-width, calc(100% - 32px));
+        max-height: var(--dialog-max-height, calc(100% - 32px));
+        border-radius: var(--dialog-border-radius, 4px);
```

Comment:
If you wanna show a dialog in 100% wide/tall (fully-fullscreen-mode), then we don't need a variable for the backdrop layer's background color. There won't be any backdrop layer anyways. 

But the fully-fullscreen-mode for a dialog doesn't make sense. A dialog **MUST** look like a dialog (a temporary layer on top of the current context). Therefore, it should be avoided, unless you want to open the dialog in a new browser tab (I guess what you refer to as popup here). In that condition (which I don't know much about and don't understand why you would need that) it's OK to show the dialog without any margins or border-radius. 

So I disagree with this change, unless you limit it somehow to popup windows.

Reply:
You can see the before and after in my images above, we want to get rid of the grey area, margins, and rounded corners. In the desktop client, all dialogs will open in new browser windows, since it will be horrible to render a dialog in the action pad. There will not be any other context behind the dialog in the same window, just an empty body

Reply:
Yes I understood that. But we already have fullscreen dialogs in the web-client since before, right? 🤔 Or do I remember incorrectly? The `.fullscreen` isn't something new, or? https://github.com/Lundalogik/lime-elements/blob/f96edc5be2cec4d0c36f1c10b04cfe3a7b73e7b9/src/components/dialog/dialog.scss#L37

My worry is that we make all those so called fullscreen dialogs suddenly look like a fullscreen app.

Reply:
Yes, but even if the dialog is set to fullscreen it will add the grey border and margin around it, something that does not make sense when the dialog is displayed alone inside its own popup window. That's why these properties are introduced

Reply:
In my first image above, the dialog is set to fullscreen without these changes. The second one is with the changes and the HTML page has set the new CSS properties

Reply:
> My worry is that we make all those so called fullscreen dialogs suddenly look like a fullscreen app.

Nothing will change unless you manually set the new properties

Reply:
> Aha.
So you mean in your example, the consumer has set all of these on 100%: `--dialog-width: 100%`, `--dialog-max-width: 100%`, `--dialog-max-height: 100%`.

> Sorry I didn't understand what had happened in the images and din't really connect the points. It'd be nice if you explained a bit more :sweat_smile: Using variables by consumers in this way is a nice solution. :ok_hand:

It sounds like you understand everything now, so I don't think I need to explain it any further :smile: This will just allow the consumer to set these properties if they want, to be able to display the dialog in a popup window